### ISSUE TITLE
pretty sure I've solved the PHP 7.1 bug but...

### ISSUE THREAD

**Issue by vesper8** [mautic/mautic#3903]
... but I don't know how to fix it short of forking this package (the culprit I believe) : https://github.com/willdurand/BazingaOAuthServerBundle

Let me preface this by saying that I really love Mautic but it's causing me some pains right now that I can't run it on my php 7.1 servers.. so I've invested about 2 hours using the debugger in phpstorm and running some tests to identify what exactly is causing the problem. I've narrowed it all down to this one line in a package that hasn't been updated in 3 years

The error when running composer install/update script:

```
[RuntimeException]
  An error occurred when executing the "'cache:clear --no-warmup'" command:

    [Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException]
    You have requested a non-existent service "kernel.debug".
```

And a similar error if you try running the installer/upgrader:

```
[error] 23948#23948: *59 FastCGI sent in stderr: "PHP message: Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: You have requested a non-existent service "kernel.debug". - in file /home/vagrant/mautic.omitted.com/vendor/symfony/dependency-injection/ContainerBuilder.php - at line 816" while reading response header from upstream, client: 192.168.10.1, server: mautic.omitted.dev, request: "GET /index.php/index.php/installer HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php7.1-fpm.sock:", host: "mautic.omitted.dev", referrer: "http://mautic.omitted.dev/index.php/installer"
```

Are caused by this line:

https://github.com/willdurand/BazingaOAuthServerBundle/blob/master/DependencyInjection/BazingaOAuthServerExtension.php#L32

```
$configuration = new Configuration($container->get('kernel.debug'));
```

I've found 3 ways to make the problem go away

```
$configuration = new Configuration($container->get('kernel.debug', 2)); # sets to NULL if service not found
```

```
$configuration = new Configuration($container->get('kernel.debug', 3)); # ignores error if service not found
```

AND the winner IMHO

```
$configuration = new Configuration($container->getParameter('kernel.debug'));
```

Why? Because I've scoured the code and the vendor folder for other references to 'kernel.debug' and there are quite a few of them (nearly two dozen of them in fact), but *ALL* of them use the ->getParameter('kernel.debug') syntax. The Bazinga package above is the *only* one to use the older ->get('kernel.debug') syntax

So there you go. I'm not sure if there's an elegant way to resolve this but since this is a 3 year old package forking it doesn't sound like such a bad idea. Unless someone knows the maintainer and could get him to change his ->get() to ->getParameter()

I would have loved to nail this with a PR but not sure how to best go about it. I hope someone with more experience can now apply a fix that will hopefully be included in 2.8.0 !!!

Referencing the other php 7.1 issues and the people involved

https://github.com/mautic/mautic/pull/3573
https://github.com/mautic/mautic/pull/3416
@escopecz @alanhartless 

Thanks guys! Keep up the great work!

**Comment by gavinerickson** [comment]
tried this out and the amendment ("winner") allowed me to install fine..

**Comment by github-grabaz** [comment]
Awesome!  Worked for me as well.  Now running mautic on most recent laravel homestead box.

**Comment by vesper8** [comment]
I too am now running 2.8.0 on a php 7.1 server. So far everything seems perfect.

Be mindful by the way that performing the upgrade 2.7.1 -> 2.8.0 leads to problems because the first step in the upgrade process fetches the zip file and then overwrites all your files which will overwrite the 3 files needing to be modified in order to work on php 7.1:

```
/upgrade.php
/app/middlewares/VersionCheckMiddleware.php
/vendor/willdurand/oauth-server-bundle/DependencyInjection/BazingaOAuthServerExtension.php
```

So inevitably the upgrade will stall/fail and you'll need to finish it manually

I ended up performing the update on a php7.0 server instead, then moved the files/db to a php7.1 server. All is good now!

Hoping a fix is patched in soon :) I noticed that @alanhartless has permissions on the willdurand/BazingaOAuthServerBundle so hopefully just a matter of time now!

**Comment by ty2u** [comment]
For what it's worth, mautic already has a fork of this package that it uses:
https://github.com/mautic/BazingaOAuthServerBundle/
https://github.com/mautic/mautic/blob/staging/composer.json#L124

**Comment by dongilbert** [comment]
@vesper8 Could you make your suggested changes as a pull request against our fork of that bundle?

**Comment by vesper8** [comment]
@dongilbert ah I was about to but looks like you beat me to it :) It's just changing get('kernel.debug') to getParameter('kernel.debug')

sorry I'm in Asia so opposite timezone I imagine.

Look forward to seeing this on staging! Thanks for getting on top of it! Been running Mautic on php 7.1 for 2 weeks now and no problems.

**Comment by vesper8** [comment]
Very happy to see this fix has been pushed to staging. It doesn't look like it was included in 2.8.1 however. 2.8.2 hopefully.

Just wondering if anyone currently running on php 7.1 and on 2.8.0 has tried to do the automatic update to 2.8.1. Does it work? Being on 7.1 on 2.7.1 and updating to 2.8.0 did not work as it overwrote files that are required for php 7.1 to work. I imagine the same thing happens when updating from 2.8.0 to 2.8.1 ?

**Comment by escopecz** [comment]
PHP 7.1 is supported.
